cabal-install/src/Distribution/Client/CmdHaddockProject.hs

```diff
@@ -254,6 +250,8 @@ haddockProjectAction flags _extraArgs globalFlags = do
           | not localStyle ->
               return []
         Left package -> do
+          -- TODO: this might not work for public packages with sublibraries
```

Comment:
This sounds a little worrying. Maybe we should preemptively create an issue describing this issue in a little more detail than it makes sense in a comment.

Reply:
Do you know a package with sublibraries published on hackage?  I could test it, and remove the TODO if it works.

Reply:
https://github.com/haskell/cabal/issues/9026

Reply:
Yeah, we are not aware of any and we are looking for a test bunny. :)

Reply:
@coot the rumor is, this is one example: https://hackage.haskell.org/package/lsp-types